pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -335,28 +326,30 @@ func (r *ReconcileKubeDirectorCluster) syncCluster(
 
 // Calculates md5sum of resource-versions of all resources
 // connected to this cluster
-func calcConnectionsHash(con *kdv1.Connections, ns string) string {
-	if con == nil {
-		return ""
-	}
+func calcConnectionsHash(
+	con *kdv1.Connections,
+	ns string) string {
 
 	clusterNames := con.Clusters
 	var buffer bytes.Buffer
 	for _, c := range clusterNames {
 		clusterObj, _ := observer.GetCluster(ns, c)
-		rv := clusterObj.ResourceVersion
-		buffer.WriteString(rv)
+		buffer.WriteString(c)
+		buffer.WriteString(strconv.Itoa(
+			int(*clusterObj.Status.SpecGenerationToProcess)))
```

Comment:
I guess technically we should worry about the possibility that this is nil; don't want to get a nil deref exception here.

If it's nil we could just use the string "nil" to represent it, should be fine.